### ISSUE TITLE
JS Code Node: "operation not permitted"

### ISSUE THREAD

**Issue by hmenzagh** [langgenius/dify#11674]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.13.2

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

TLDR: Running Python3 Code nodes works fine, JS one outputs error "operation not permitted" regardless of the code.
Details:
- I'm running Dify self-hosted with docker-compose. Pretty stock & vanilla at this point.
- My machine is a Apple Silicon mac running Asahi Linux, so platform is ideally linux/arm64.
- Tried changing the sandbox container's platform to amd64 & arm64
- There are no errors in the logs, just a log attesting POST request is recieved
- Sandbox API Key is personalised
- Error appears by running the node or the workflow itself
- Error is not linked to input params either
- Tried using this image too: langgenius/dify-sandbox:f40de1f6bc5f87d0e847cbf52076280bf61c05d5
Any ideas ? 😦

### ✔️ Expected Behavior

Not an error

### ❌ Actual Behavior

Returns error: "operation not permitted"

**Comment by yihong0618** [comment]
also found that in my local machine (docker-compose mac m3) will give a try to figure out why these days

**Comment by yihong0618** [comment]
more info
if we run ` pytest -s api/tests/integration_tests/workflow`

the javascript side fails 

```cli
_______________________________________________________________ test_javascript_plain ________________________________________________________________

    def test_javascript_plain():
        code = 'console.log("Hello World")'
>       result_message = CodeExecutor.execute_code(language=CODE_LANGUAGE, preload="", code=code)

api/tests/integration_tests/workflow/nodes/code_executor/test_code_javascript.py:12: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'core.helper.code_executor.code_executor.CodeExecutor'>, language = <CodeLanguage.JAVASCRIPT: 'javascript'>, preload = '', code = 0

    @classmethod
    def execute_code(cls, language: CodeLanguage, preload: str, code: str) -> str:
        """
        Execute code
        :param language: code language
        :param code: code
        :return:
        """
        url = URL(str(dify_config.CODE_EXECUTION_ENDPOINT)) / "v1" / "sandbox" / "run"
    
        headers = {"X-Api-Key": dify_config.CODE_EXECUTION_API_KEY}
    
        data = {
            "language": cls.code_language_to_running_language.get(language),
            "code": code,
            "preload": preload,
            "enable_network": True,
        }
    
        try:
            response = post(
                str(url),
                json=data,
                headers=headers,
                timeout=Timeout(
                    connect=dify_config.CODE_EXECUTION_CONNECT_TIMEOUT,
                    read=dify_config.CODE_EXECUTION_READ_TIMEOUT,
                    write=dify_config.CODE_EXECUTION_WRITE_TIMEOUT,
                    pool=None,
                ),
            )
            if response.status_code == 503:
                raise CodeExecutionError("Code execution service is unavailable")
            elif response.status_code != 200:
                raise Exception(
                    f"Failed to execute code, got status code {response.status_code},"
                    f" please check if the sandbox service is running"
                )
        except CodeExecutionError as e:
            raise e
        except Exception as e:
            raise CodeExecutionError(
                "Failed to execute code, which is likely a network issue,"
                " please check if the sandbox service is running."
                f" ( Error: {str(e)} )"
            )
    
        try:
            response = response.json()
        except:
            raise CodeExecutionError("Failed to parse response")
    
        if (code := response.get("code")) != 0:
            raise CodeExecutionError(f"Got error code: {code}. Got error msg: {response.get('message')}")
    
        response = CodeExecutionResponse(**response)
    
        if response.data.error:
>           raise CodeExecutionError(response.data.error)
E           core.helper.code_executor.code_executor.CodeExecutionError: error: operation not permitted

api/core/helper/code_executor/code_executor.py:116: CodeExecutionError
________________________________________________________________ test_javascript_json ________________________________________________________________

    def test_javascript_json():
        code = dedent("""
        obj = {'Hello': 'World'}
        console.log(JSON.stringify(obj))
        """)
>       result = CodeExecutor.execute_code(language=CODE_LANGUAGE, preload="", code=code)

api/tests/integration_tests/workflow/nodes/code_executor/test_code_javascript.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'core.helper.code_executor.code_executor.CodeExecutor'>, language = <CodeLanguage.JAVASCRIPT: 'javascript'>, preload = '', code = 0

    @classmethod
    def execute_code(cls, language: CodeLanguage, preload: str, code: str) -> str:
        """
        Execute code
        :param language: code language
        :param code: code
        :return:
        """
        url = URL(str(dify_config.CODE_EXECUTION_ENDPOINT)) / "v1" / "sandbox" / "run"
    
        headers = {"X-Api-Key": dify_config.CODE_EXECUTION_API_KEY}
    
        data = {
            "language": cls.code_language_to_running_language.get(language),
            "code": code,
            "preload": preload,
            "enable_network": True,
        }
    
        try:
            response = post(
                str(url),
                json=data,
                headers=headers,
                timeout=Timeout(
                    connect=dify_config.CODE_EXECUTION_CONNECT_TIMEOUT,
                    read=dify_config.CODE_EXECUTION_READ_TIMEOUT,
                    write=dify_config.CODE_EXECUTION_WRITE_TIMEOUT,
                    pool=None,
                ),
            )
            if response.status_code == 503:
                raise CodeExecutionError("Code execution service is unavailable")
            elif response.status_code != 200:
                raise Exception(
                    f"Failed to execute code, got status code {response.status_code},"
                    f" please check if the sandbox service is running"
                )
        except CodeExecutionError as e:
            raise e
        except Exception as e:
            raise CodeExecutionError(
                "Failed to execute code, which is likely a network issue,"
                " please check if the sandbox service is running."
                f" ( Error: {str(e)} )"
            )
    
        try:
            response = response.json()
        except:
            raise CodeExecutionError("Failed to parse response")
    
        if (code := response.get("code")) != 0:
            raise CodeExecutionError(f"Got error code: {code}. Got error msg: {response.get('message')}")
    
        response = CodeExecutionResponse(**response)
    
        if response.data.error:
>           raise CodeExecutionError(response.data.error)
E           core.helper.code_executor.code_executor.CodeExecutionError: error: operation not permitted

api/core/helper/code_executor/code_executor.py:116: CodeExecutionError
_________________________________________________________ test_javascript_with_code_template _________________________________________________________

    def test_javascript_with_code_template():
>       result = CodeExecutor.execute_workflow_code_template(
            language=CODE_LANGUAGE,
            code=JavascriptCodeProvider.get_default_code(),
            inputs={"arg1": "Hello", "arg2": "World"},
        )

api/tests/integration_tests/workflow/nodes/code_executor/test_code_javascript.py:26: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
api/core/helper/code_executor/code_executor.py:138: in execute_workflow_code_template
    raise e
api/core/helper/code_executor/code_executor.py:136: in execute_workflow_code_template
    response = cls.execute_code(language, preload, runner)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'core.helper.code_executor.code_executor.CodeExecutor'>, language = <CodeLanguage.JAVASCRIPT: 'javascript'>, preload = '', code = 0

    @classmethod
    def execute_code(cls, language: CodeLanguage, preload: str, code: str) -> str:
        """
        Execute code
        :param language: code language
        :param code: code
        :return:
        """
        url = URL(str(dify_config.CODE_EXECUTION_ENDPOINT)) / "v1" / "sandbox" / "run"
    
        headers = {"X-Api-Key": dify_config.CODE_EXECUTION_API_KEY}
    
        data = {
            "language": cls.code_language_to_running_language.get(language),
            "code": code,
            "preload": preload,
            "enable_network": True,
        }
    
        try:
            response = post(
                str(url),
                json=data,
                headers=headers,
                timeout=Timeout(
                    connect=dify_config.CODE_EXECUTION_CONNECT_TIMEOUT,
                    read=dify_config.CODE_EXECUTION_READ_TIMEOUT,
                    write=dify_config.CODE_EXECUTION_WRITE_TIMEOUT,
                    pool=None,
                ),
            )
            if response.status_code == 503:
                raise CodeExecutionError("Code execution service is unavailable")
            elif response.status_code != 200:
                raise Exception(
                    f"Failed to execute code, got status code {response.status_code},"
                    f" please check if the sandbox service is running"
                )
        except CodeExecutionError as e:
            raise e
        except Exception as e:
            raise CodeExecutionError(
                "Failed to execute code, which is likely a network issue,"
                " please check if the sandbox service is running."
                f" ( Error: {str(e)} )"
            )
    
        try:
            response = response.json()
        except:
            raise CodeExecutionError("Failed to parse response")
    
        if (code := response.get("code")) != 0:
            raise CodeExecutionError(f"Got error code: {code}. Got error msg: {response.get('message')}")
    
        response = CodeExecutionResponse(**response)
    
        if response.data.error:
>           raise CodeExecutionError(response.data.error)
E           core.helper.code_executor.code_executor.CodeExecutionError: error: operation not permitted

api/core/helper/code_executor/code_executor.py:116: CodeExecutionError
```

**Comment by yihong0618** [comment]
> Additional Info:
> 
> * Works on my M2 mac running MacOS
> * Does not work on Asahi Linux (tried a fresh install)
> * Works on my RPi

I build local(use orbstack) and load in my M3 macbook it succeed...

**Comment by hmenzagh** [comment]
Just tried on 15.2 from a fresh install, still failing
